src/sqlfluff/core/rules/std/L015.py

```diff
@@ -30,10 +30,42 @@ def _eval(self, segment, raw_stack, **kwargs):
 
         Look for DISTINCT keyword immediately followed by open parenthesis.
         """
-        # We only trigger on start_bracket (open parenthesis)
-        if segment.name == "start_bracket":
-            filt_raw_stack = self.filter_meta(raw_stack)
-            if len(filt_raw_stack) > 0 and filt_raw_stack[-1].name == "DISTINCT":
-                # If we find DISTINCT followed by open_bracket, then bad.
-                return LintResult(anchor=segment)
-        return LintResult()
+        # We only trigger on expressions that begin with start_bracket (open
+        # parenthesis)
+        if segment.type == "expression":
+            segments_filt = self.filter_meta(segment.segments)
+            if (
+                len(segments_filt)
+                and segments_filt[0].type == "start_bracket"
+                and segments_filt[0].raw == "("
+            ):
+                filt_raw_stack = self.filter_meta(raw_stack)
+                if len(filt_raw_stack) and filt_raw_stack[-1].name == "DISTINCT":
+                    # If we find open_bracket immediately following DISTINCT,
+                    # then bad.
+                    fixes = []
+                    if len(segments_filt) >= 2:
```

Comment:
I wonder if this check could (should?) be more sophisticated, like don't add a space if there's already a space present. Maybe it doesn't even need to be conditional, but I felt safer checking the length, to avoid a potential `IndexError` when creating the `LintFix`.